cmd/config.go

```diff
@@ -191,6 +191,8 @@ type Config struct {
 	AllowedSigningAlgos *AllowedSigningAlgos
 
 	SubscriberAgreementURL string
+
+	Features map[string]bool
```

Comment:
> Per the config plan, this should be a type that is embedded into the config struct of each component that uses it.

I put this at the top level as otherwise it couldn't be loaded in `AppShell.Run` without knowing _which_ service was being run, which seemed like kind of a pain in the ass.

> Also, why string rather than a struct? The latter is nice because you get checking for typos, and you can be sure that once a flag is gone, it's gone. I guess the string types are a way of skirting circular dependencies between config and the various components?

Mostly circular dependencies. I used a `error`/`panic` in `Set`/`Enabled` as kind of a safe guard against typos (but it's true something like that might not ever actually get triggered in testing) but it'd also be possible to write a super simple `go/parser` tool to check that no feature is misspelled/unused etc and add it to our tests.

Also using a `struct` (or `int` etc) would require a bunch of, most likely generated, code to help with string conversion etc for exporting the state via `expvar`.